GitCommands/Submodules/SubmoduleStatusProvider.cs

```diff
@@ -170,12 +168,15 @@ private async Task GetSubmoduleStatusAsync(bool updateStatus, SubmoduleInfo info
                 return;
             }
 
+            // sequential prefetching FilesEncoding in a parent thread
+            var foo = supermodule.FilesEncoding;
```

Comment:
After this PR it is at least responsive, but there is no GitModules caching. On every refersh GitModules are created for each submodule and have to load the same data (like git dir) again and again. I think we should observe submodules for modifications and refresh only these submodules which have changed. The same way as we use file system watchers. Otherwise if there is a lots of submodules the refresh action queries for changes for each submodule causing overall system overloading and cpu/hdd wasting.

Reply:
File status reloading: #6359 (will not handle the case when all submodules are changed)
Diff loading: See #5769 Git-status already provides information about which submodules that are changed. Has been awaiting the other submodule changes, sidepanel etc

Reply:
Thanks for referencing the issues, my knowledge about the project is little outdated.

Reply:
It is great to have you around, of course I have tried to do this myself too but it takes me quite some time to get into this.